common/util.go

```diff
@@ -387,6 +387,13 @@ func IsNotFoundError(err error) bool {
 	return errors.As(err, &notFoundErr)
 }
 
+func ErrorHash(err error) string {
+	if err != nil {
+		return fmt.Sprintf("0x%x", farm.Fingerprint32([]byte(err.Error())))
```

Comment:
nitpick: we're basically turning a string into an opaque string. the fact that it's hashing it to an integer and encoding it as hex is an implementation detail. so the `0x` prefix is not useful and kind of misleading. I would just do this if you want a fixed length:
```suggestion
		return fmt.Sprintf("%08x", farm.Fingerprint32([]byte(err.Error())))
```

Reply:
agree